api/api.ts

```diff
@@ -0,0 +1,244 @@
+/*---------------------------------------------------------------------------------------------
```

Comment:
Just wanted to check... I was expecting we'd want to have a separate repo for this similar to the cpptools API.  I'm assuming that this is being added to the CMake Tools repo temporarily for convenience.  Is that correct?

Reply:
I was originally thinking that the API type definitions can be hosted and published directly from this repo (it's currently structured as a separate package in the same repo), but if you'd rather the API live in a separate repo I'm not opposed.

Reply:
Yes, I'd rather have it be separate.  But we can do that after you've implemented the CMake Tools side of things.

Reply:
Changed to pull from published NPM package in latest update.